nifi-nar-bundles/nifi-influxdb-bundle/nifi-influxdb-processors/src/main/java/org/apache/nifi/processors/influxdb/ExecuteInfluxDBQuery.java

```diff
@@ -0,0 +1,199 @@
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        super.onScheduled(context);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+        if ( flowFile.getSize() == 0) {
+            getLogger().error("Empty query");
+            flowFile = session.putAttribute(flowFile, INFLUX_DB_ERROR_MESSAGE, "Empty query size is " + flowFile.getSize());
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        Charset charset = Charset.forName(context.getProperty(CHARSET).evaluateAttributeExpressions(flowFile).getValue());
+        String database = context.getProperty(DB_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        TimeUnit queryResultTimeunit =
+            TimeUnit.valueOf(context.getProperty(INFLUX_DB_QUERY_RESULT_TIMEUNIT).evaluateAttributeExpressions(flowFile).getValue());
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        session.exportTo(flowFile, baos);
+        String query = new String(baos.toByteArray(), charset);
+
+        try {
+            long startTimeMillis = System.currentTimeMillis();
+            QueryResult result = executeQuery(context, database, query, queryResultTimeunit);
+            String json = gson.toJson(result);
+
+            if ( getLogger().isDebugEnabled() ) {
+                getLogger().debug("Query result {} ", new Object[] {result});
+            }
+
+            ByteArrayInputStream bais = new ByteArrayInputStream(json.getBytes(charset));
+            session.importFrom(bais, flowFile);
+            final long endTimeMillis = System.currentTimeMillis();
+
+            if ( ! result.hasError() ) {
+                flowFile = session.putAttribute(flowFile, INFLUX_DB_EXECUTED_QUERY, String.valueOf(query));
+                session.transfer(flowFile, REL_SUCCESS);
+            } else {
+                flowFile = populateErrorAttributes(session, flowFile, query, result.getError());
+                session.transfer(flowFile, REL_FAILURE);
+            }
+
+            session.getProvenanceReporter().send(flowFile, new StringBuilder("influxdb://")
+                .append(context.getProperty(INFLUX_DB_URL).evaluateAttributeExpressions().getValue()).append("/")
+                .append(database).toString(),
+                (endTimeMillis - startTimeMillis));
+        } catch (Exception exception) {
+            flowFile = populateErrorAttributes(session, flowFile, query, exception.getMessage());
+            if ( exception.getCause() instanceof SocketTimeoutException ) {
```

Comment:
Are there any other network exceptions that need to be caught here?

Reply:
I did not find any documentation on any other network exception on invoking query.  But let me know if you have any recommendations.